pkg/apis/navigator/v1alpha1/types.go

```diff
@@ -32,38 +32,58 @@ type CassandraCluster struct {
 type CassandraClusterSpec struct {
 	NavigatorClusterConfig `json:",inline"`
 
-	NodePools []CassandraClusterNodePool `json:"nodePools"`
+	// List of node pools belonging to this CassandraCluster.
+	// NodePools cannot currently be removed.
+	// A cluster with 0 node pools will not function correctly.
+	NodePools []CassandraClusterNodePool `json:"nodePools,omitempty"`
 
-	// Image describes the database image to use
-	Image *ImageSpec `json:"image"`
+	// Image describes the Cassandra database image to use.
+	// This should only be set if version auto-detection is not desired.
+	// If set, the image tag used must correspond to the version specified
+	// in 'spec.version'.
+	Image *ImageSpec `json:"image,omitempty"`
 
 	// The version of the database to be used for nodes in the cluster.
+	// This field must be a valid Cassandra version, e.g. '3.11.2'.
 	Version version.Version `json:"version"`
 }
 
 // CassandraClusterNodePool describes a node pool within a CassandraCluster.
 type CassandraClusterNodePool struct {
-	Name     string `json:"name"`
-	Replicas int32  `json:"replicas"`
+	// Name of the node pool specified as a DNS_LABEL.
+	// Each node pool must specify a name.
+	Name string `json:"name"`
+
+	// The number of desired replicas in this node pool.
+	// This value will correspond to the number of replicas in the created
+	// StatefulSet.
+	// If not set, a default of 1 will be used.
+	Replicas *int32 `json:"replicas,omitempty"`
 
-	// Persistence specifies the configuration for persistent data for this
-	// node.
+	// Persistence specifies the persistent volume configuration for this node.
+	// Disabling persistence can cause issues when a node restarts.
+	// Cannot be updated.
+	// +optional
 	Persistence PersistenceConfig `json:"persistence,omitempty"`
 
 	// NodeSelector should be specified to force nodes in this pool to run on
 	// nodes matching the given selector.
+	// In future this may be superceded by an 'affinity' field.
 	// +optional
-	NodeSelector map[string]string `json:"nodeSelector"`
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
 
 	// Rack specifies the cassandra rack with which to label nodes in this
-	// nodepool. If this is not set, a default will be selected.
+	// nodepool.
+	// If this is not set, the name of the node pool will be used.
 	// +optional
-	Rack string `json:"rack"`
+	Rack string `json:"rack,omitempty"`
```

Comment:
Correct - I had figured there'd be no harm in putting omitempty onto this field was all.

This is a greater issue when validation is performed client side afaict (because when it is, if the value is set to "" then it won't actually be sent in the JSON payload). Given kubectl doesn't actually know about these types however, I don't think it makes a difference in our case. Happy to remove it.